readers/common/src/main/java/com/criteo/hadoop/garmadon/reader/GarmadonReader.java

```diff
@@ -11,20 +13,37 @@
 import org.slf4j.LoggerFactory;
 
 import java.io.ByteArrayInputStream;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
 import java.nio.ByteBuffer;
 import java.util.*;
 import java.util.concurrent.CompletableFuture;
 
 public class GarmadonReader {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(GarmadonReader.class);
+    public static final String CONSUMER_PREFIX_NAME = "garmadon.reader";
+
+    public static String HOSTNAME;
+
+    static {
+        try {
+            HOSTNAME = InetAddress.getLocalHost().getCanonicalHostName();
+        } catch (UnknownHostException e) {
+            LOGGER.error("", e);
```

Comment:
Then hostname is null which is dangerous for a public field. Either we set a default value or we must exit(1) if we cannot get it.